### DISCUSSION TITLE
how to upgrade Gradio in Stable Diffusion

### DISCUSSION THREAD

**Discussion by pardoart** [AlUlkesh/stable-diffusion-webui-images-browser#204] [Q&A] [answered]
First of all, sorry for my ignorance. My knowledge about programming and computers is very simple, because I am an illustrator. I am trying to install the extension in Stable Diffusion and I get the error:

_You are running Gradio version 3.16.2. This version of the extension requires at least Gradio version 3.23.0._

I try to update it with Anaconda Powershell with the following sequence:

cd C:\Userstable-diffusion-webui

.\venv "scripts" activate

pip install gradio --upgrade

It seems to be installed, but I still get the error and in the folder _C:\Userserstable-diffusion-webui.\venvibsite-packages_ is still the folder "_gradio-3.16.2.dist-info_".

Can anyone help me with some information to update Gradio inside Stable Diffusion?

Thank you very much

**Comment by AlUlkesh** [comment]
Well, I have a few thoughts about that.

1. It's possible that you are running an old version of automatic1111 webui. When started, it would reinstall the gradio-version it expects. So try to update that (git pull - if you need help with that, let me know)
2. If that is not the case, you seem to run both Anaconda and venv, which in this case try to do the same thing (make a separate environment from other python apps). It's possible there are side effects with that. Try running without Anaconda.
3. If it still doesn't work, you can delete the venv folder. It will be recreated on the next run.
4. And if all fails, you can try a fresh install of automatic1111 in a different folder.

If you neeed help with any of the steps, feel free to ask.

**Reply by AlUlkesh** [reply]
Oh - and make sure these lines in webui-user.bat look like that, with no parameters:
```
set PYTHON=
set GIT=
set VENV_DIR=
```

**Reply by pardoart** [reply]
Thank you very much for the help. I think I have updated A1111 since I have "git pull" in "webui-user.bat" Is that correct? My understanding is that with that line, it updates to the latest version. I don't know how to see if I have the latest version in the GUI. In the extensions tab, inside "installed", I have updated all the versions of the extensions.

at the bottom of the GUI appears:  python: 3.10.6  •  torch: 1.13.1+cu117  •  xformers: N/A  •  gradio: 3.16.2 

I deleted the Venv folder and reinstalled the 3.16.2 folder.

I'm sorry, but I don't understand point 2. As I told you, I don't have an advanced level with these topics...

I have written the lines in the .bat correctly
set PYTHON=
set GIT=
set VENV_DIR=

I have this:
@echo off

set PYTHON=
set GIT=
set VENV_DIR=
set COMMANDLINE_ARGS=--gradio-img2img-tool color-sketch --api
git pull
call webui.bat

Maybe the set COMMANDLINE_ARGS line is wrong. 

Thank you so much, as I told you before.

**Reply by AlUlkesh** [reply]
When you start webui-user.bat, the first thing it prints out, should be something like this:

```
venv "c:\tools\Ai\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:50:30) [MSC v.1933 64 bit (AMD64)]
Version: v1.3.2-219-ga009fe15
Commit hash: a009fe15fd98b44aede07c47bf7938fb4280924b
```
How do these lines look in your case?

**Reply by pardoart** [reply]
This is what I get:

fatal: not a git repository (or any of the parent directories): .git
venv "C:\Users\User\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Commit hash: <none>
Installing requirements for Web UI


Launching Web UI with arguments: --gradio-img2img-tool color-sketch --api
No module 'xformers'. Proceeding without it.
2023-06-09 12:04:43,017 - ControlNet - INFO - ControlNet v1.1.220
ControlNet preprocessor location: C:\Users\User\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads
2023-06-09 12:04:43,250 - ControlNet - INFO - ControlNet v1.1.220
Image Browser: ImageReward is not installed, cannot be used.
Loading weights [79e42fb744] from C:\Users\User\stable-diffusion-webui\models\Stable-diffusion\perfectWorld_v2Baked.safetensors
Creating model from config: C:\Users\User\stable-diffusion-webui\configs\v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Loading VAE weights specified in settings: C:\Users\User\stable-diffusion-webui\models\VAE\vae-ft-ema-560000-ema-pruned.ckpt
Applying cross attention optimization (Doggettx).
Textual inversion embeddings loaded(4): charturnerv2, easynegative, Trigger_Studio, ulzzang-6500-v1.1
Textual inversion embeddings skipped(1): 21charturnerv2
Model loaded in 9.1s (load weights from disk: 0.2s, create model: 0.5s, apply weights to model: 4.8s, apply half(): 0.7s, load VAE: 0.5s, move model to device: 0.8s, load textual inversion embeddings: 1.5s).
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.

**Reply by AlUlkesh** [reply]
This seems to be the problem:

`fatal: not a git repository (or any of the parent directories): .git`

For some reason it can't find your git information. So any `git pull` won't work and you probably have an old version of the webui.

Let's see if we can figure out what is going on. First, can you do this:

Open a cmd window (WindowsKey + R and enter `cmd`) and there enter:

```
cd C:\Users\User\stable-diffusion-webui
dir .git
```
What does it say at this point?

**Reply by pardoart** [reply]
I get this (translated from Spanish):

File not found

C:\Userstable-diffusion-webui>dir .git
 The volume on drive C has no label.
 The volume serial number is: XXXX-XXXX

directory of C:\Users\User\stable-diffusion-webui

File not found

C:\Userserstable-diffusion-webui> directory


-

In the stable-diffusion- webui I have a txtfile called ".gitignore"... Is it right?

Thank you so much for your help

**Reply by AlUlkesh** [reply, marked as answer]
Yes, .gitignore is correct.

Unfortunately, your .git directory seems to have been lost somehow, and that can't be reconstructed. I highly suggest you make a fresh install.

Luckily by now that's easy and quick if you follow the right steps. Here's how I would do it:

(For the following I am assuming the directory where you currently installed webui is `C:\Users\User`)

Open a cmd window (WindowsKey + R and enter cmd) and there enter:

```
cd C:\Users\User
git clone https://github.com/AUTOMATIC1111/stable-diffusion-webui sd-webui2
```
This will create a new webui directory for you, called `sd-webui2`

Now go ahead and copy or move your old model files over, that way you don't have to re-download them.

All files from
`C:\Users\User\stable-diffusion-webui\models\Stable-diffusion`
to
`C:\Users\User\sd-webui2\models\Stable-diffusion`

Then in cmd:

```
cd C:\Users\User\sd-webui2
webui-user.bat
```
This might take a while on the first start, since every needed package is being download.

When webui starts, go ahead and close it down again.

Now you can copy or move over the remaining relevant files from your old version, that is these folders:

```
embeddings
extensions
outputs
log
models\ControlNet
models\Esrgan
models\hypernetworks
models\Lora
models\VAE
```
Then start the new webui again with `webui-user.bat`

At this point everything should work again. If you changed any of the settings, you'll have to do that again.

Hope this helps.

**Reply by pardoart** [reply]
Yes!!! Thank you so much. It works perfectly. After copying the models, lora, extensions, outpus ect.... I guess I can delete the old directory, can't I?

**Reply by AlUlkesh** [reply]
Great, glad it worked.

Well, if it was me I'd rename the old directory and wait a few days to see, if everything is working. Then I'd delete it.

**Reply by pardoart** [reply]
You've been so kind. Thank you for your help. I'll do it and I hope this post helps someone with the problem I had.

Gracias, amigo!!!
